delfin/drivers/huawei/oceanstor/oceanstor.py

```diff
@@ -354,6 +354,146 @@ def list_disks(self, context):
+                consts.FS_WORM_COMPLIANCE: constants.WORMType.COMPLIANCE,
+                consts.FS_WORM_AUDIT_LOG: constants.WORMType.AUDIT_LOG,
+                consts.FS_WORM_ENTERPRISE: constants.WORMType.ENTERPRISE
+            }
+            for fs in fss:
+                status = constants.FilesystemStatus.FAULTY
+                if fs['HEALTHSTATUS'] == consts.FS_HEALTH_NORMAL:
+                    status = constants.FilesystemStatus.NORMAL
+                type = constants.FSType.THICK
+                if fs['ALLOCTYPE'] == consts.FS_TYPE_THIN:
+                    type = constants.FSType.THIN
+
+                sector_size = int(fs['SECTORSIZE'])
+                total_cap = int(fs['CAPACITY']) * sector_size
+                used_cap = int(fs['ALLOCCAPACITY']) * sector_size
+
+                compressed = False
+                if fs['ENABLECOMPRESSION'] != 'false':
+                    compressed = True
+
+                deduplicated = False
+                if fs['ENABLEDEDUP'] != 'false':
+                    deduplicated = True
+
+                fs = {
+                    'name': fs['NAME'],
+                    'storage_id': self.storage_id,
+                    'native_filesystem_id': fs['ID'],
+                    'compressed': compressed,
+                    'deduplicated': deduplicated,
+                    'worm': worm_type.get(fs['WORMTYPE'],
+                                          constants.WORMType.NON_WORM),
+                    'status': status,
+                    'type': type,
+                    'total_capacity': total_cap,
+                    'used_capacity': used_cap,
+                    'free_capacity': None,
+                }
+                fs_list.append(fs)
+
+            return fs_list
+
+        except exception.DelfinException as err:
+            err_msg = "Failed to get filesystem metrics from OceanStor: %s" %\
+                      (six.text_type(err))
+            LOG.error(err_msg)
+            raise err
+
+        except Exception as err:
+            err_msg = "Failed to get filesystem metrics from OceanStor: %s" %\
+                      (six.text_type(err))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+
+    def list_qtrees(self, context):
+        try:
+            # Get list of OceanStor qtrees details
+            qts = self.client.get_all_qtrees()
+            security_mode = {
+                consts.SECURITY_STYLE_MIXED: constants.NASSecurityMode.MIXED,
+                consts.SECURITY_STYLE_NATIVE: constants.NASSecurityMode.NATIVE,
+                consts.SECURITY_STYLE_NTFS: constants.NASSecurityMode.NTFS,
+                consts.SECURITY_STYLE_UNIX: constants.NASSecurityMode.UNIX,
+            }
+
+            qt_list = []
+            for qt in qts:
+                fs_id = None
+                if qt['PARENTTYPE'] == consts.PARENT_OBJECT_TYPE_FS:
+                    fs_id = qt['PARENTID']
+                qt = {
```

Comment:
confirm whther path can be filled or not?

Reply:
This will be handled in separate issue